### DISCUSSION TITLE
Is there a HOCON lexer?

### DISCUSSION THREAD

**Discussion by Andre601** [pygments/pygments#2822] [Q&A]
I'm looking for a hocon (`.conf` files) lexer, but I'm unable to find one...
Does Pygments offer one (maybe under different name), or would I need to make one. And if I had to make one, does there already one exist by the community?

Apreciate the help.

**Comment by birkenfeld** [comment]
It would seem not - of course many config file formats are very similar to each other so you might be lucky with an existing one...

**Reply by Andre601** [reply]
The main issue is actually finding it, which given the amount of pre-existing lexers Pygment has, may not be an easy task...

I probs could it limit to just config-related lexers first and then expand if nothing is found...
Still will take time, so I hoped there may be a community-managed HOCON lexer available.

**Reply by Andre601** [reply]
Going through documents-based lexers, the closest ones seem to be Lighty and Unixconfig. Both have their pros and cons in different ways:

- Lighty:
  - `+` `key=value` is properly separated by colors
  - `-` Numbers, including in keys, are highlighted a different color.
- Unixconfig:
  - `+` Seems to highlight characters like =, {, [, etc. properly.
  - `-` It's... bland. key remains one color while `=value` is all one color.

I probably stick with Lighty as it is more visually distinct here... Maybe one day will I find a lexer that fits better.
